### PR TITLE
video_core: Avoid setting alpha tev with Dot3_RGBA8 color op

### PR DESCRIPTION
**Pull request by GPUCode**
* Further reduces unnecessary shader regenerations in games like ALBW and SM3L

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6907)

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_shader_gen.cpp [hunk 1]
```diff
@@ -228,6 +233,9 @@ PicaFSConfig PicaFSConfig::BuildFromRegs(const Pica::Regs& regs, bool use_normal
                              FramebufferRegs::FragmentOperationMode::Shadow;
 
     state.shadow_texture_orthographic = regs.texturing.shadow.orthographic != 0;
+    if (state.shadow_rendering) {
+        state.shadow_texture_orthographic.Assign(regs.texturing.shadow.orthographic != 0);
```
Comment (Steveice10) [inline src/video_core/renderer_opengl/gl_shader_gen.cpp:237]: Is the line above this supposed to be removed? It looks like we set it twice now.

File: src/video_core/renderer_vulkan/vk_shader_gen.cpp [hunk 2]
```diff
@@ -232,6 +237,9 @@ PicaFSConfig::PicaFSConfig(const Pica::Regs& regs, const Instance& instance) {
                                   FramebufferRegs::FragmentOperationMode::Shadow);
 
     state.shadow_texture_orthographic.Assign(regs.texturing.shadow.orthographic != 0);
+    if (state.shadow_rendering) {
+        state.shadow_texture_orthographic.Assign(regs.texturing.shadow.orthographic != 0);
```
Comment (Steveice10) [inline src/video_core/renderer_vulkan/vk_shader_gen.cpp:241]: Ditto here

**Review by Steveice10 — APPROVED** [review]

**Comment by GPUCode** [comment]
Some context behind the changes. When Dot3_RGBA is used all alpha ops are [ignored](https://github.com/citra-emu/citra/blob/e129f070471310c94f4a44f8f44059f86039b129/src/video_core/renderer_opengl/gl_shader_gen.cpp#L618). Because this operation is very common it ends up generating a bunch of duplicate shaders, so the alpha parameters are masked out to avoid that.

**Comment by m4xw** [comment]
Can confirm this change not only reduces the shaders but also improves overall performance in some games up to 8-10% which i suspect might be excessive switching between duped shaders, in particular in albw / in front of links house.
I didn't expect that one, great stuff.

**Pull request merged**